### DISCUSSION TITLE
Flask Upload Misdirects File Upload in One Directory Mode

### DISCUSSION THREAD

**Discussion by aadejare** [pyinstaller/pyinstaller#8386] [Help]
This message is an addendum to this issue I had [here](https://github.com/pyinstaller/pyinstaller/issues/8248) 
Creating a flask application, I have specific folders that the files should upload to which are bundled in the One Directory Mode of Pyinstaller. While running the application, it will upload the files, but once the application closes, the files upload, but to the present directory of the EXE and not directory it should be located in.  How can I adjust the runtime directory to point to the directory folder that will be built during pyinstaller?

**Comment by bwoodsend** [comment]
> I have specific folders that the files should upload to which are bundled in the One Directory Mode 

Well that's invalid behaviour so the answer is put it somewhere else.

**Reply by aadejare** [reply]
The application itself runs with files from the one directory and relies on uploads into the same directory. I want to know if it's feasible and if not why is it invalid behavor?

**Reply by rokm** [reply]
> The application itself runs with files from the one directory and relies on uploads into the same directory. I want to know if it's feasible and if not why is it invalid behavor?

It sounds like you are implicitly relying on current working directory being the program's directory. Which is not the case with PyInstaller-frozen applications (unless you use one-dir with `--content-directory="."`, which removes the `_internal` subdirectory). It would not work with unfrozen python application either, if you ran it from an arbitrary working directory. 

If you want to rely on assumed working directory, you will need to ensure that it is set to what you expect it to be (e.g., do a `os.chdir(sys._MEIPASS)` or `os.chdir(os.path.dirname(__file__))` at the start of your program).

Otherwise, you should set your application's upload directories using absolute paths, anchored to `sys._MEIPASS` or `os.path.dirname(__file__)`, as suggested by [documentation](https://pyinstaller.org/en/stable/runtime-information.html#using-file). (Note that `sys._MEIPASS` is PyInstaller-specific, while `os.path.dirname(__file__)` should work both in unfrozen python script and in frozen application).
